test/util/OperationTestHelpers.h

```diff
@@ -85,4 +85,54 @@ class ShallowParentOperation : public Operation {
   }
 };
 
+class AlwaysFailLazyOperation : public Operation {
+  std::vector<QueryExecutionTree*> getChildren() override { return {}; }
+  string getCacheKeyImpl() const override { return "AlwaysFailLazyOperation"; }
+  string getDescriptor() const override {
+    return "AlwaysFailLazyOperationDescriptor";
+  }
+  size_t getResultWidth() const override { return 0; }
+  size_t getCostEstimate() override { return 0; }
+  uint64_t getSizeEstimateBeforeLimit() override { return 0; }
+  float getMultiplicity([[maybe_unused]] size_t) override { return 0; }
+  bool knownEmptyResult() override { return false; }
+  vector<ColumnIndex> resultSortedOn() const override { return {}; }
+  VariableToColumnMap computeVariableToColumnMap() const override { return {}; }
+
+ public:
+  using Operation::Operation;
+  ProtoResult computeResult([[maybe_unused]] bool requestLaziness) override {
+    return {[]() -> cppcoro::generator<IdTable> {
+              throw std::runtime_error{"AlwaysFailLazyOperation"};
+              // Required so that the exception only occurs within the generator
+              co_return;
+            }(),
+            resultSortedOn(), LocalVocab{}};
+  }
+};
+
+class CustomGeneratorOperation : public Operation {
+  cppcoro::generator<IdTable> generator_;
+  std::vector<QueryExecutionTree*> getChildren() override { return {}; }
+  string getCacheKeyImpl() const override { return "CustomGeneratorOperation"; }
```

Comment:
At least make the cache key random, else you might get strange caching behavior once this is reused.